v5.16.0 errors on yarn install

### What happened?

I have upgraded to version 5.16.0 and got an error after running `yarn upgrade-interactive`



### Example

`yarn upgrade interactive`

then upgrade to 5.16.0 version
```
➤ YN0000: ┌ Link step
➤ YN0007: │ @pulumi/vault@npm:5.16.0 must be built because it never has been before or the last one failed
➤ YN0009: │ @pulumi/vault@npm:5.16.0 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-59412d6c/build.log)
```

```
cat /tmp/xfs-59412d6c/build.log                           
# This file contains the result of Yarn building a package (@pulumi/vault@npm:5.16.0)
# Script name: install

node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/home/abukamel/codebases/work/iac/pulumi/programs/deployment/node_modules/@pulumi/vault/scripts/install-pulumi-plugin.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.18.0
```

### Output of `pulumi about`

```
CLI          
Version      3.89.0
Go Version   go1.21.1
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in nodejs: executable='/home/abukamel/.nvm/versions/node/v18.18.0/bin/node' version='v18.18.0'
```

### Additional context

Downgrading to `v5.15.1` resolved the issue
### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/.github/workflows/master.yml
+++ b/.github/workflows/master.yml
@@ -307,7 +307,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Publish SDKs
-      uses: pulumi/pulumi-package-publisher@v0.0.12
+      uses: pulumi/pulumi-package-publisher@v0.0.9
     - env:
         SLACK_CHANNEL: provider-upgrade-publish-status
         SLACK_COLOR: "#FF0000"

--- a/.github/workflows/prerelease.yml
+++ b/.github/workflows/prerelease.yml
@@ -253,7 +253,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Publish SDKs
-      uses: pulumi/pulumi-package-publisher@v0.0.12
+      uses: pulumi/pulumi-package-publisher@v0.0.9
     - env:
         SLACK_CHANNEL: provider-upgrade-publish-status
         SLACK_COLOR: "#FF0000"

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -265,7 +265,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Publish SDKs
-      uses: pulumi/pulumi-package-publisher@v0.0.12
+      uses: pulumi/pulumi-package-publisher@v0.0.9
     - env:
         SLACK_CHANNEL: provider-upgrade-publish-status
         SLACK_COLOR: "#FF0000"

